### PR TITLE
Enable app module doctest discovery (#1761)

### PR DESCRIPTION
**Pull request by noah-aviel-dove**
See my comments on [the issue](https://github.com/DataBiosphere/azul/issues/1761) for an explanation of how this addresses the issue.

### CODE REVIEW

**Comment by noah-aviel-dove** [comment]
@hannes-ucsc's original instructions were to leave the module name unspecified to allow the default to persist, however `importlib.util.spec_from_file_location` requires an explicitly provided name for the module, so to keep things simple I'm just explicitly providing the "default" name

File: src/azul/modules.py [hunk 1]
```diff
@@ -39,4 +39,5 @@ def load_module(path: str,
 
 def load_app_module(lambda_name, **module_attributes):
     path = os.path.join(config.project_root, 'lambdas', lambda_name, 'app.py')
-    return load_module(path, '__main__', module_attributes)
+    # Changing the module name here will break doctest disscoverability
```
Comment (hannes-ucsc) [inline src/azul/modules.py:42]:
```suggestion
    # Changing the module name here will break doctest discoverability
```

**Comment by noah-aviel-dove** [comment]
Commit show out of order for some reason. The correct sequence is:
```
$ git log --oneline
0fd70216 (HEAD -> issues/noah-aviel-dove/1761-app-module-doctests, origin/issues/noah-aviel-dove/1761-app-module-doctests) drop! Build failure in service app doctests
baee4f08 Use correct __name__ when loading app module (#1761)
e3fca547 Remove modules with no doctests from test_doctests
9ad3038d Verify doctest discovery
```

**Comment by jessebrennan** [comment]
> Commits display out of order for some reason
Github UI orders commits by author date instead of topologically for some reason. See: https://github.com/isaacs/github/issues/386

**Pull request merged**